connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorMaker.java

```diff
@@ -0,0 +1,243 @@
+ *
+ * Each Herder represents a separate source->target replication flow.
+ */
+public class MirrorMaker {
+    private static final Logger log = LoggerFactory.getLogger(MirrorMaker.class);
+
+    private static final long SHUTDOWN_TIMEOUT_SECONDS = 60L;
+
+    private static final List<Class> CONNECTOR_CLASSES = Arrays.asList(
+        MirrorSourceConnector.class,
+        MirrorHeartbeatConnector.class,
+        MirrorCheckpointConnector.class);
+ 
+    private final Map<SourceAndTarget, Herder> herders = new HashMap<>();
+    private CountDownLatch startLatch;
+    private CountDownLatch stopLatch;
+    private final AtomicBoolean shutdown = new AtomicBoolean(false);
+    private final ShutdownHook shutdownHook;
+    private final String advertisedBaseUrl;
+    private final Time time;
+    private final MirrorMakerConfig config;
+
+    public MirrorMaker(MirrorMakerConfig config, Time time) {
+        log.debug("Kafka MirrorMaker instance created");
+        this.time = time;
+        this.advertisedBaseUrl = "TODO";
+        this.config = config;
+        config.clusterPairs().forEach(x -> addHerder(x));
+        shutdownHook = new ShutdownHook();
+    }
+
+    public MirrorMaker(Map<String, String> props, Time time) {
+        this(new MirrorMakerConfig(props), time);
+    }
+
+    public MirrorMaker(Map<String, String> props) {
+        this(props, Time.SYSTEM);
+    }
+
+    public void start() {
+        log.info("Kafka MirrorMaker starting with {} herders.", herders.size());
+        if (startLatch != null) {
+            throw new IllegalStateException("MirrorMaker instance already started");
+        }
+        startLatch = new CountDownLatch(herders.size());
+        stopLatch = new CountDownLatch(herders.size());
+        Runtime.getRuntime().addShutdownHook(shutdownHook);
+        for (Herder herder : herders.values()) {
+            try {
+                herder.start();
+            } finally {
+                startLatch.countDown();
+            }
+        }
+        log.info("Configuring connectors...");
+        config.clusterPairs().forEach(x -> configureConnectors(x));
+        log.info("Kafka MirrorMaker started");
+    }
+
+    public void stop() {
+        boolean wasShuttingDown = shutdown.getAndSet(true);
+        if (!wasShuttingDown) {
+            log.info("Kafka MirrorMaker stopping");
+            for (Herder herder : herders.values()) {
+                try {
+                    herder.stop();
+                } finally {
+                    stopLatch.countDown();
+                }
+            }
+            log.info("Kafka MirrorMaker stopped.");
+        }
+    }
+
+    public void awaitStop() {
+        try {
+            stopLatch.await();
+        } catch (InterruptedException e) {
+            log.error("Interrupted waiting for MirrorMaker to shutdown");
+        }
+    }
+
+    public void pause(String source, String target) {
```

Comment:
`pause()` and `resume()` methods are unused. And, you can merge both these methods by taking action as a parameter.

Reply:
We can drop these, thanks.